src/core/hooks/use-is-builder-ready.ts

```diff
@@ -0,0 +1,9 @@
+import { atom, useAtom } from "jotai";
+
+export const isBuilderReadyAtom = atom<boolean>(false);
+
+// * Hook to identify if the builder is ready
+export const useIsBuilderReady = () => {
```

Comment:
we should rename it to `useIsPageLoaded()`
builder is is ready mode if we switch pages